### ISSUE TITLE
Visualize leaf criterion

### ISSUE THREAD

**Issue by tlapusan** [parrt/dtreeviz#81]
hi @parrt,

leaves from regressor and classifier trees contains two important information : number of samples and criterion.
Criterion for regressor are “mse”, “friedman_mse”, “mae” and for classifer are "gini" and "entropy". All of them shows the leaf performance/confidence for its predictions.

I think would be helpful to have some visualisations for leaf criterion. Each leaf performance, in the end, will determine the general tree performance.
It's also useful to interpret leaf criterion in parallel with leaf samples. For ex. a leaf could have a very good performance (ex. gini -> 0), but that leaf contains very few samples... so that leaf will overfit. The best scenario for a leaf is to have good performance and many leaf samples.

<img width="879" alt="Screen Shot 2020-03-23 at 5 36 07 PM" src="https://user-images.githubusercontent.com/12815158/77334063-fdaa3380-6d2c-11ea-9cff-8851bb742d7f.png">

We can also plot the histogram of leaf criterion. This is useful when the tree has a lot of leaves or when you want to have a general point-of-view about all leaf criterion (general tree performance).

<img width="870" alt="Screen Shot 2020-03-23 at 5 41 36 PM" src="https://user-images.githubusercontent.com/12815158/77334508-917bff80-6d2d-11ea-8c06-93744a8c84c5.png">

I already have an implementation for these visualisations (95% ready for a PR).
What do you think about it ? :)

**Comment by tlapusan** [comment]
hi @parrt, any feedback on this ? :) thanks

**Comment by parrt** [comment]
Sorry for the delay. Been focusing on paper.  if I understand correctly, we already have a histogram of number of samples but you are proposing adding histogram showing the criterion?

**Comment by tlapusan** [comment]
@parrt, yes, that's correct. I've already mentioned the reasons why I think these new visualisations  can add new values/insights :)

**Comment by parrt** [comment]
I think this is useful. (btw, We should be careful to make sure that our histograms always have bar width that are the same for consistency, just like they are in the decision nodes.)

I wonder if we want a combined diagram where we multiply the number of samples by the criterion?  Maybe both? Then, at a glance, people can look at the histogram to see which ones have the most confidence?

**Comment by tlapusan** [comment]
I like the idea of multiplying number of samples by criterion. I will make a plot to see how it looks.

I don't know what you mean by "Maybe both?" :)

**Comment by parrt** [comment]
Maybe we want a separate histogram to show the criterion and the ability to show the multiplication of the two.  Actually, if the criterion is an error metric, then we should be dividing by the error, since that will give a bigger bar if the error is low.

**Comment by tlapusan** [comment]
hi @parrt,

I started to think at visualisations about sample/criterion for leaves.
But when I put it on paper, something happened :). 
Let's say we have two leaves, with 10 and 100 samples. Criterion 0.1 and 1 (for entropy, 1 is the worst)
based on our above formula :
leaf1  10/0.1 = 100 confidence
leaf2  100/1 = 100 confidence

OR another example :
leaf1 10/0.5 = 20
leaf2 20/0.5 = 40  (here these two leaves should have the same confidence)

I still like the idea of combining both #samples and criterion, but we need to think to another formula. Until then, I've created a PR with visualisations mentioned in the beginning of this issue.
